transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -367,9 +429,23 @@ impl SignatureRctBulletproofs {
 
         // Each MLSAG must be valid.
         for (i, ring) in rings.iter().enumerate() {
+            let rules_digest = ring
+                .input_rules
+                .as_ref()
+                .map(|rules| rules.digest())
+                .unwrap_or_default();
+
+            // If there are input rules, then the signature is over the rules digest
+            // If not, then it is the entire extended message digest
```

Comment:
Can you please elaborate on why this is the case?

Reply:
I am guessing because thats how the ring MLSAG was generated when it was presigned, but I am wondering if it shouldn't include more than just the rules?

Reply:
i mean the short answer is that it's by design of the MCIP -- i'd be inclined to just add a code comment like "For more information about the rationale see discussion in the MCIP"